src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageServer.cs

```diff
@@ -151,6 +151,11 @@ public static Task<RazorLanguageServer> CreateAsync(Stream input, Stream output,
                         services.AddSingleton<AdhocWorkspaceFactory, DefaultAdhocWorkspaceFactory>();
                         services.AddSingleton<ProjectSnapshotChangeTrigger>((services) => services.GetRequiredService<GeneratedDocumentPublisher>());
 
+                        services.AddSingleton<WorkspaceSemanticTokensRefreshPublisher, DefaultWorkspaceSemanticTokensRefreshPublisher>();
+
+                        services.AddSingleton<DefaultWorkspaceSemanticTokensRefreshTrigger, DefaultWorkspaceSemanticTokensRefreshTrigger>();
```

Comment:
Why do you need to register it as a `DefaultWorkspaceSemanticTokensRefreshTrigger` isn't registering it as a `ProjectSnapshotChangeTrigger` below sufficient?